packages/zqlite/src/table-source.ts

```diff
@@ -512,26 +519,30 @@ export class TableSource implements Source {
  * This applies all filters present in the AST for a query to the source.
  * This will work until subquery filters are added
  * at which point either:
- * a. we move optional filters to connect
- * b. we do the transform of removing subquery filters from optionalFilters while
+ * a. we move filters to connect
+ * b. we do the transform of removing subquery filters from filters while
  *    preserving the meaning of the filters.
  *
  * https://www.notion.so/replicache/Optional-Filters-OR-1303bed895458013a26ee5aafd5725d2
  */
-export function optionalFiltersToSQL(filters: NoSubqueryCondition): SQLQuery {
+export function filtersToSQL(filters: NoSubqueryCondition): SQLQuery {
   switch (filters.type) {
     case 'simple':
       return simpleConditionToSQL(filters);
     case 'and':
-      return sql`(${sql.join(
-        filters.conditions.map(condition => optionalFiltersToSQL(condition)),
-        sql` AND `,
-      )})`;
+      return filters.conditions.length > 0
```

Comment:
The need to handle the empty cases was revealed by dropping where conditions when they were fully applied by the source.